dia/ideas2.rkt

```diff
@@ -0,0 +1,57 @@
+#lang racket
+
+(provide (rename-out [mb #%module-begin]) #%datum #%top #%top-interaction #%app)
+
+#| (Unstable) Module language for ideas
+
+Format:
+  #lang abe/ideas2 exported-ideas-id
+
+  - Ideas appraisal tree [N%]
+
+Meaning:
+
+- The body is an "appraisal" tree. Roughly, markdown bulleted list in tree
+form, with bracketed percentags [N%] at the end of the line.
+- Validate appraisals.
+- Tally idea attributions.
+- Validate attributions.
+- Export (provide) the exported-ideas-id bound to the attributions.
+
+|#
+
+(require syntax/parse/define
+         abe/dia)
+
+(define-syntax-parse-rule
+  (mb export:id {~datum <=} tree-data:expr)
```

Comment:
Are we still using the `<=` syntax? I don't see it used in the patch.

Reply:
This shows up in the (syntax) reader's output. Try `raco read <new lang file>` (_e.g._, `evaluation/antecedents/ideas.md` after applying the patch to change languages).

I get
```
(module ideas abe/antecedents2
  (#%module-begin
   antecedents-attributions
   <=
   ("../appraisal/ideas.md" ideas)
   <=
   #hash(("arrows" . ("Haskell" "Hughes 2000"))
         ("embeddable language" . ("Racket" "Rash" "Symex"))
         ("extensible syntax"
          .
          ("Racket" "Scheme" "Lisp" "Rash" "Ballantyne et al 2020"))
         ("flow-oriented semantics"
          .
          ("Unix pipelines"
           "Threading macro"
           "pipeR"
           "Hsiao-Peng et. al. 1999"))
         ("functional/composable programming"
          .
          ("Scheme" "Lisp" "ML" "Haskell" "Clojure" "Racket"))
         ("immutable data" . ("Clojure" "Haskell" "Scala" "Erlang"))
         ("multi-valued I/O" . ("Verilog" "VHDL" "Scheme" "Racket"))
         ("phrase structure" . ("APL" "Vim"))
         ("point-free programming" . ("APL" "Haskell" "Symex"))
         ("structural debugging" . ("Debug"))
         ("sunyata logic" . ("Kasivajhula 2014-"))
         ("variadic functions"
          .
          ("Racket" "Lisp" "Scheme" "Python" "Ruby" "Clojure"))
         ("visual syntax"
          .
          ("APL" "Scratch (MIT)" "Threading macro" "pipeR")))))
```

You can also do `git grep '<='` to see
```
antecedents2.rkt:26:25:  (mb export:id {~datum <=} [ideas-mod:expr imported-ideas:id] {~datum <=} tree-data:expr)
antecedents2.rkt:61:30:    (strip-context #`(#,name <= #,import <= #,tree)))
attribution2.rkt:32:25:  (mb export:id {~datum <=} tree-data:expr)
attribution2.rkt:60:30:    (strip-context #`(#,name <= #,tree)))
ideas2.rkt:27:25:  (mb export:id {~datum <=} tree-data:expr)
ideas2.rkt:55:30:    (strip-context #`(#,name <= #,tree)))
```

The `strip-context` calls are producing the output of the syntax readers; the result is processed by the corresponding module-begin macro.

It also means that, if you really wanted to, you could write the `(module ideas abe/antecedents2 antecedents-attributions <= ("../appraisal/ideas.md" ideas) <= …)` instead (not recommended).